packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts

```diff
@@ -0,0 +1,226 @@
+/**
+ * This file is for Compatibility Matrix tests, which assigns specific
+ * Allure context tags.
+ */
+
+import * as allure from 'allure-vitest'
+import { pipe } from 'fp-ts/function'
+import { describe, expect, test } from 'vitest'
+import { clientFactory, setupPeerTestsLifecycle } from '../util'
+import { annotateAllure } from "./util";
+import { datamodel, sugar } from '@iroha2/data-model'
+import { CLIENT_CONFIG } from '@iroha2/test-configuration'
+import { Seq } from 'immutable'
+
+setupPeerTestsLifecycle()
+
+// Read Allure API: https://allurereport.org/docs/vitest-reference/
+describe('Compatibility Matrix tests', () => {
+  // Based on https://github.com/AlexStroke/iroha-java/blob/007a9ac55991cd8a2b519e62a10144156d9f8301/modules/client/src/test/kotlin/jp/co/soramitsu/iroha2/InstructionsTest.kt#L134
+  test('Register domain', async (ctx) => {
+    await annotateAllure(ctx, {
+      id: 4075,
+      feature: 'Domain',
+      story: 'Account registers a domain',
+      sdkTestId: 'register_domain',
+    })
+
+    const DOMAIN_NAME = 'new_domain_name'
+    let result
+    let final
```

Comment:
What is the purpose of defining `final` here?

Define it right where you need it unless you don't need to move a value from a nested scope; and use `const` if you aren't going to re-assign.

And please read this article very carefully: http://dmitrysoshnikov.com/ecmascript/javascript-the-core-2nd-edition-rus/

Reply:
ready